lightning-invoice/src/lib.rs

```diff
@@ -85,14 +85,12 @@ mod sync;
 
 pub use de::{ParseError, ParseOrSemanticError};
 
-// TODO: fix before 2037 (see rust PR #55527)
-/// Defines the maximum UNIX timestamp that can be represented as `SystemTime`. This is checked by
-/// one of the unit tests, please run them.
-const SYSTEM_TIME_MAX_UNIX_TIMESTAMP: u64 = core::i32::MAX as u64;
+/// The number of bits used to represent timestamps as defined in BOLT 11.
+const TIMESTAMP_BITS: usize = 35;
 
-/// Allow the expiry time to be up to one year. Since this reduces the range of possible timestamps
-/// it should be rather low as long as we still have to support 32bit time representations
-const MAX_EXPIRY_TIME: u64 = 60 * 60 * 24 * 356;
+/// The maximum timestamp that can be represented as a [`Duration`] since the UNIX epoch while
+/// allowing for adding an expiry without overflowing.
+const MAX_TIMESTAMP: u64 = core::u64::MAX >> (64 - TIMESTAMP_BITS);
```

Comment:
a bit confusing.  isn't `(1 << TIMESTAMP_BITS) - 1` more idiomatic?

also, comment says "allowing for adding an expiry" - what is the max expiry time and should we subtract it from the above?

Reply:
> a bit confusing. isn't `(1 << TIMESTAMP_BITS) - 1` more idiomatic?
> 

Yes, thanks! I temporarily used that in another PR in the context of log approximation, but had blanked on the idiom after dropping the commit. 😛 

> also, comment says "allowing for adding an expiry" - what is the max expiry time and should we subtract it from the above?

We dropped the maximum expiry, simply allowing anything that fits into a `u64` and using a saturating add if we ever overflow. The `u64` space is enormous, so it should never be a problem in reality. Updated the comment.